enhancements/sig-policy/54-core-policy-module/README.md

```diff
@@ -0,0 +1,227 @@
+
+<!--
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). 
+
+All code is expected to have sufficient e2e tests.
+-->
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+<!--
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- [Maturity levels][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, stable), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+-->
+
+### Upgrade / Downgrade Strategy
+
+New features and adjustments to existing features should be made in a backwards
+compatible way. In particular, this means that new items in common structs
+to be embedded by policy types can not be required.
+
+Since the intent is to have other policy controllers import this for common
+functions, those functions should maintain their behavior, except for bugfixes.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+Each piece of the policy framework could expose types/methods for behavior that
+that piece requires. For example, the status-sync could expose a function for
+creating the events that it is watching for, and the template-sync could expose
+a type with the common fields it requires like `remediationAction`. In this
+case, the policy controllers would need to import each piece of the framework,
+and keep each updated to match each other if changes were made. There would
+also not be a clear place for non-required, but commonly used fields like the
+`spec.namespaceSelector`, and their behavior.
+
+## Infrastructure Needed
+
+A new repository, name TBD. Maybe `policy-framework`?
```

Comment:
Let's not take that name in case we upstream the stolostron policy-framework repo.

How about `policy-controller-runtime` or `policy-controller-core`?

Reply:
The stolostron policy-framework repo is (almost entirely) tests and documentation. IMO, we could merge that content into this community repo if we wanted to "upstream" it.

I would prefer something without "core" - that would lead to stuttering in the type names. For example, in the current proposal I have `policycore.PolicyCoreSpec`, which I don't like. Something like `framework.PolicyCoreSpec` would make me happier. Alternatively, if we changed the name of `PolicyCoreSpec`, that could help.

Both `policy-controller-runtime` and `policy-controller-core` feel too close to well-known k8s libraries `sigs.k8s.io/controller-runtime` and `k8s.io/api/core/v1`.